src/shims/unix/foreign_items.rs

```diff
@@ -58,6 +58,16 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
             // 512 seems to be a reasonable default. The value is not critical, in
             // the sense that getpwuid_r takes and checks the buffer length.
             ("_SC_GETPW_R_SIZE_MAX", |this| Scalar::from_int(512, this.pointer_size())),
+            // the machine's fd table does not start with a pre-allocated size but just the
+            // standard streams so we can't use it here and the spec imposes
+            // a minimum of `_POSIX_OPEN_MAX` (20).
+            ("_SC_OPEN_MAX", |this| Scalar::from_int(20, this.pointer_size())),
```

Comment:
20 seems really really low here. Why would we report such a low limit?

Reply:
I can set to 1024 (what linux returns) some other platforms return higher values. wdyt ?

Reply:
Isn't it dynamic on Linux, determined by `ulimit`?

Reply:
1024 being the max here indeed. but then 20 is to low for you so what do you propose ?

Reply:
The `ulimit` can easily be much higher, like 64k.

Reply:
on illumos it s 65536 on my debian it is only 1024 by default with unlimited but as root I can push it higher indeed.

Reply:
But I guess 1024 is not an unreasonable value. The comment should make it clear that this is completely arbitrary and we can adjust if it that turns out to be helpful.